src/mmrelay/meshtastic_utils.py

```
@@ -956,14 +1414,23 @@ def _get_device_metadata(client: Any) -> dict[str, Any]:
+            return result
+
         # Preflight: client may be a mock without localNode/getMetadata
-        if not getattr(client, "localNode", None) or not hasattr(
-            client.localNode, "getMetadata"
+        if not getattr(client, "localNode", None) or not callable(
+            getattr(client.localNode, "getMetadata", None)
         ):
+            if raise_on_error:
+                raise RuntimeError(
+                    "Meshtastic client has no localNode.getMetadata() for metadata probe"
+                )
```

Ruff (0.15.4)

[warning] Abstract `raise` to an inner function

(TRY301)